app/src/main/java/ch/epfl/sdp/blindly/match/UserListFilter.kt

```diff
@@ -0,0 +1,105 @@
+package ch.epfl.sdp.blindly.match
+
+import android.location.Location
+import android.os.Build
+import androidx.annotation.RequiresApi
+import ch.epfl.sdp.blindly.user.User
+import java.lang.IllegalArgumentException
+
+private const val ONE_KM_IN_METERS = 1000
+
+class UserListFilter {
+    /**
+     * Filters a list of users, so that only the ones that are in the location and age range of
+     * the current user remain.
+     *
+     * @param currentUser the current user
+     * @param otherUsers the list of other users
+     * @return the filtered list of users
+     */
+    @RequiresApi(Build.VERSION_CODES.O)
+    fun filterLocationAndAgeRange(currentUser: User, otherUsers: List<User>): List<User> {
+        val filteredList: MutableList<User> = ArrayList<User>().toMutableList()
+        val minAge = currentUser.ageRange?.get(0)
+        val maxAge = currentUser.ageRange?.get(1)
+
+        for (user in otherUsers) {
+            val otherUserAge = User.getUserAge(user)
+            if (otherUserAge != null) {
+                if (otherUserAge in minAge!!..maxAge!! && isLocatedInUserRadius(
+                        currentUser,
+                        user
+                    )
+                ) {
+                    filteredList += user
+                }
+            }
+        }
+
+        return filteredList
+    }
+
+    /**
+     * Does the same checks than [MatchingAlgorithm.getPotentialMatchesFromDatabase], but from the
```

Comment:
If they are doing the same checks wouldn't it be possible to merge these two and simply reverse the order of arguments passed. Though not sure if that would break anything, correct functioning comes first. I leave it up to you.

Reply:
Unfortunately, it wouldn't be possible since `getPotentialMatchesFromDatabase` queries a Firebase Firestore database (see in `MatchingAlgorithm.kt`), which is not possible in this class (because it uses a list of users instead).

Reply:
I see, just wanted to check. Thanks for looking into it.